src/NuGet/Microsoft.Net.Compilers.Toolset/buildMultiTargeting/Microsoft.Net.Compilers.Toolset.props

```diff
@@ -0,0 +1,4 @@
+﻿<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project>
+  <Import Project="..\build\Microsoft.Net.Compilers.Toolset.props" />
```

Comment:
Why is this file needed? Compilation happens in the inner build.

Reply:
I'm following the guidelines here: https://natemcmaster.com/blog/2017/07/05/msbuild-task-in-nuget/ If those are incorrect I can remove. As is though it appears to be working based off the success of the bootstrap phase.

@natemcmaster

Reply:
If Nick is right and there is no task or target needed in the outer build of a multitargeting project, then this file can be removed. 

Related note: NuGet just added support for another folder, buildTransitive/. If you want  this package's targets to be imported when a project has a transitive reference to this package, add a file just like this one to that folder.

Reply:
I think we want to avoid the transitive on for now. One "feature" of this package is you can install it on a single project to temporarily work around a compiler bug. The isolation there to that project is intended. It's possible that's not actually needed anymore but for the moment intent is to keep it (want to minimize the impact of this change). 

> If Nick is right and there is no task or target needed in the outer build of a multitargeting project, then this file can be removed.

Can either of you elaborate on this a bit? I'm not sure why this would or wouldn't be needed. I trust you two are right here but would like to understand this a bit better if possible.

Reply:
There are two different builds / evaluations of the the project that we call "inner build" and "outer build". In the outer build, TargetFramework is unset and build loops over the TargetFrameworks and performs individual inner builds with the target framework set. 

build/PackageName.props|targets is imported into inner builds only
buildMulltitarget/PackageName.props|targets is imported into outer builds only

Some of the confusion here may have to do with the fact that you aware multi-targeting the task, which makes `buildMultitargeting` seem relevant. However, this folder is not about the multi-targeting of the tasks in the package, but the multi-targeting of the projects that reference the package.

There's no one universal approach to deciding what to put in build/ vs buildMultitargeting/, it depends on what the task does and how it wants to achieve it.

1. Same code in inner build as outer build (what you have here by having buildMultitargeting turn around and import from build/)
2. Different code in inner and outer build (using different files or what you have + some conditions to handle the two cases differently).
3. Some code in inner build, no code at all in outer build
4. Some code in outer build, no code at all in inner build

Some things can just ignore that the outer build exists as they are purely features of the inner build. Compilation is like that. There's no CoreCompile in the outer build, only in the inner build where we are actually producing the assembly for a single target framework. Other things do have to handle the outer build in some way. Pack is a good example of that: we need a target that produces a single .nupkg that aggregates all the inner build outputs.